appservice/src/main/java/nu/yona/server/device/rest/DeviceController.java

```diff
@@ -397,18 +460,30 @@ public DeviceResource toResource(DeviceBaseDto device)
 			ControllerLinkBuilder selfLinkBuilder = getSelfLinkBuilder(device.getId());
 			addSelfLink(selfLinkBuilder, deviceResource);
 			addEditLink(selfLinkBuilder, deviceResource);
+			if (isRequestingDevice(device))
```

Comment:
❓ [Initially](https://wiki.yona.nu/display/DEV/Adding+device+notion?focusedCommentId=18284704#comment-18284704), you were doubting if we should move the links at all. What is the reason that you think it is better to move them after all? It introduces some complexity to requesting the devices collection and marking the requesting device.

Reply:
It's a cleaner design. The information is really device specific, so I feel it's better to put the information in the context it belongs. It's more clear to the app builders.